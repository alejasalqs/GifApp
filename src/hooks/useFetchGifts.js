import { useState, useEffect } from 'react';
import { getGifts } from '../Helpers/getGifts';

export const useFetchGifts = ( category ) => {

    const [state, setState] = useState({
        data: [],
        loading: true
    });

      // Los efectos no pueden ser async
      // Nos ayuda a ejecutar código de manera condicional
      useEffect(() => {
        // Primer argumento, funcion que vamos a ejecutar
        getGifts(category).then(gifts => setState({
            data: gifts,
            loading: false
        }));
    }, 
    // Segundo argumento, Si se envía vacío solo se ejecuta la primera vez que se renderiza el componente
    // Si se añade algo al arreglo significa que se ejecuta si la condicion se cumple
    [category]);
    
    return state;
}