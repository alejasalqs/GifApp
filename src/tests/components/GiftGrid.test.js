const { shallow } = require("enzyme");
import '@testing-library/jest-dom';
const { GiftGrid } = require("../../components/GiftGrid");
import React from 'react';
import { useFetchGifts } from '../../hooks/useFetchGifts';
jest.mock('../../hooks/useFetchGifts');

describe('Pruebas en el archivo GiftGrid.js', () => {
    
    const category = 'Pokemon'

    test('El componenete debe renderizarse correctamente ', () => {

        useFetchGifts.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GiftGrid category={category}/>)
    
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes ', () => {
        // mock => fingir algo

        const fakeGifs = {
            id: 'ABC', 
            title: 'TEST',
            url: 'https://test.jpg',
        }

        useFetchGifts.mockReturnValue({
            data: [fakeGifs],
            loading: false
        });
        
        const wrapper = shallow( <GiftGrid category={category}/>)
        
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GiftGridItem').length ).toBe( fakeGifs.length );

    });
    
})
