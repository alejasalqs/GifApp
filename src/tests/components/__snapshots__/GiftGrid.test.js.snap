// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo GiftGrid.js Debe de mostrar items cuando se cargan imagenes  1`] = `
<Fragment>
  <h3>
    Searching: 
    Pokemon
  </h3>
  <div
    className="card-grid"
  >
    <GiftGridItem
      id="ABC"
      key="ABC"
      title="TEST"
      url="https://test.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el archivo GiftGrid.js El componenete debe renderizarse correctamente  1`] = `
<Fragment>
  <h1>
    Loading...
  </h1>
  <h3>
    Searching: 
    Pokemon
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
